Error occurred while parsing an SNMP message

Not sure what is causing this. I am doing a fair few SNMP requests per 5 sec and 60 sec. Let me know if you need more information. I should also add this happens infrequently but enough to be annoying.

Woops! An error occurred while parsing an SNMP message. :(
To have this problem corrected, please report the information below verbatim
via email to snmp@nym.se or by creating a GitHub issue at
https://github.com/calmh/node-snmp-native/issues

Thanks!

AssertionError: 48 == 66
    at parse (C:\Users\crb\Desktop\snmp\node_modules\snmp-native\lib\snmp.js:228:12)
    at EventEmitter.msgReceived (C:\Users\crb\Desktop\snmp\node_modules\snmp-native\lib\snmp.js:410:15)
    at emitTwo (events.js:106:13)
    at Socket.emit (events.js:191:7)
    at UDP.onMessage (dgram.js:548:8)

Message data:
    42 00 00 00 05 33 f0 2c d1 1f c5 41 00 97 b0 32
    e2 1d 10 5b a4 c0 a8 40 80 ca 6f 01 00 00 00 0f
    00 57 00 49 00 4e 00 2d 00 43 00 4c 00 36 00 35
    00 33 00 32 00 47 00 45 00 44 00 42 00 48

events.js:160
      throw er; // Unhandled 'error' event
      ^
AssertionError: 48 == 66
    at parse (C:\Users\crb\Desktop\snmp\node_modules\snmp-native\lib\snmp.js:228:12)
    at EventEmitter.msgReceived (C:\Users\crb\Desktop\snmp\node_modules\snmp-native\lib\snmp.js:410:15)
    at emitTwo (events.js:106:13)
    at Socket.emit (events.js:191:7)
    at UDP.onMessage (dgram.js:548:8)

Another one:

Woops! An error occurred while parsing an SNMP message. :(
To have this problem corrected, please report the information below verbatim
via email to snmp@nym.se or by creating a GitHub issue at
https://github.com/calmh/node-snmp-native/issues

Thanks!

AssertionError: 48 == 83
    at parse (C:\Users\crb\Desktop\snmp\node_modules\snmp-native\lib\snmp.js:228:12)
    at EventEmitter.msgReceived (C:\Users\crb\Desktop\snmp\node_modules\snmp-native\lib\snmp.js:426:15)
    at emitTwo (events.js:106:13)
    at Socket.emit (events.js:191:7)
    at UDP.onMessage (dgram.js:548:8)

Message data:
    53 70 6f 74 55 64 70 30 3b 42 e7 c7 16 0a 89 c7
    00 01 00 04 48 95 c2 03 54 94 14 6e f1 c2 ec f7
    60 92 ae a5 45 46 04 76 74 b5 ae 8b

events.js:160
      throw er; // Unhandled 'error' event
      ^
AssertionError: 48 == 83
    at parse (C:\Users\crb\Desktop\snmp\node_modules\snmp-native\lib\snmp.js:228:12)
    at EventEmitter.msgReceived (C:\Users\crb\Desktop\snmp\node_modules\snmp-native\lib\snmp.js:426:15)
    at emitTwo (events.js:106:13)
    at Socket.emit (events.js:191:7)
    at UDP.onMessage (dgram.js:548:8)

--- a/lib/snmp.js
+++ b/lib/snmp.js
@@ -410,7 +410,7 @@ function msgReceived(msg, rinfo) {
     try {
         pkt = parse(msg);
     } catch (error) {
-        return self.parseError(error, msg);
+        return self.emit('error', error);
     }
 
     // If this message's request id matches one we've sent,
@@ -521,35 +521,6 @@ Session.prototype.requestId = function () {
     }
 
     return ((now & 0x1fffff) << 10) + self.counter;
-};
-
-// Display useful debugging information when a parse error occurs.
-
-Session.prototype.parseError = function (error, buffer) {
-    var self = this, hex;
-
-    // Display a friendly introductory text.
-    console.error('Woops! An error occurred while parsing an SNMP message. :(');
-    console.error('To have this problem corrected, please report the information below verbatim');
-    console.error('via email to snmp@nym.se or by creating a GitHub issue at');
-    console.error('https://github.com/calmh/node-snmp-native/issues');
-    console.error('');
-    console.error('Thanks!');
-
-    // Display the stack backtrace so we know where the exception happened.
-    console.error('');
-    console.error(error.stack);
-
-    // Display the buffer data, nicely formatted so we can replicate the problem.
-    console.error('\nMessage data:');
-    hex = buffer.toString('hex');
-    while (hex.length > 0) {
-        console.error('    ' + hex.slice(0, 32).replace(/([0-9a-f]{2})/g, '$1 '));
-        hex = hex.slice(32);
-    }
-
-    // Let the exception bubble upwards.
-    self.emit('error', error);
 };
 
 // Send a message. Can be used after manually constructing a correct Packet structure.

--- a/lib/snmp.js
+++ b/lib/snmp.js
@@ -438,16 +438,6 @@ function msgReceived(msg, rinfo) {
 
             entry.callback(null, pkt.pdu.varbinds);
         }
-    } else {
-        // This happens if we receive the response to a message we've already timed out
-        // and removed the request entry for. Maybe we shouldn't even log the warning.
-
-        // Calculate the approximate send time and how old the packet is.
-        var age = (Date.now() & 0x1fffff) - (pkt.pdu.reqid >>> 10);
-        if (age < 0) {
-            age += 0x200000;
-        }
-        console.warn('Response with unknown request ID from ' + rinfo.address + '. Consider increasing timeouts (' + age + ' ms old?).');
     }
 }
 

